pkg/cmd/roachtest/tests/restore.go

```diff
@@ -833,50 +827,86 @@ func makeHardwareSpecs(override hardwareSpecs) hardwareSpecs {
 }
 
 var defaultBackupSpecs = backupSpecs{
-	// TODO(msbutler): write a script that automatically finds the latest versioned fixture for
-	// the given spec and a reasonable aost.
+	// TODO(msbutler): write a script that automatically finds the latest versioned fixture.
 	version:          "v22.2.0",
+	cloud:            spec.AWS,
 	backupProperties: "inc-count=48",
 	fullBackupDir:    "LATEST",
-
-	// restoring as of from the 24th incremental backup in the chain
-	aost:     "'2022-12-21 05:15:00'",
-	workload: tpceRestore{customers: 25000},
+	incsIncluded:     12,
+	workload:         tpceRestore{customers: 25000},
 }
 
 type backupSpecs struct {
 	// version specifies the crdb version the backup was taken on.
 	version string
 
+	// cloud is the cloud storage provider the backup is stored on.
+	cloud string
+
+	// backupProperties identifies specific backup properties included in the backup fixture
+	// path.
 	backupProperties string
 
 	// specifies the full backup directory in the collection to restore from.
 	fullBackupDir string
 
-	// aost specifies the as of system time restore to.
-	aost string
+	// specifies the number of incrementals to restore from
+	incsIncluded int
 
 	// workload defines the backed up workload.
 	workload backupWorkload
+
+	// aost specifies the aost to restore from. Derived at runtime.
+	aost string
 }
 
-// String returns a stringified version of the backup specs. If full is false,
-// default backupProperties are omitted. Note that the backup version, backup
-// directory, and AOST are never included.
+// String returns a stringified version of the backup specs. Note that the
+// backup version, backup directory, and AOST are never included.
 func (bs backupSpecs) String(full bool) string {
 	var builder strings.Builder
 	builder.WriteString("/" + bs.workload.String())
 
 	if full || bs.backupProperties != defaultBackupSpecs.backupProperties {
 		builder.WriteString("/" + bs.backupProperties)
 	}
+	builder.WriteString("/" + bs.cloud)
+
+	if full || bs.incsIncluded != defaultBackupSpecs.incsIncluded {
+		builder.WriteString("/" + fmt.Sprintf("%d", bs.incsIncluded))
+	}
 	return builder.String()
 }
 
+func (bs backupSpecs) storagePrefix() string {
+	if bs.cloud == spec.AWS {
+		return "s3"
+	}
+	return "gs"
+}
+
+func (bs backupSpecs) backupCollection() string {
+	return fmt.Sprintf(`'%s://cockroach-fixtures/backups/%s/%s/%s?AUTH=implicit'`,
+		bs.storagePrefix(), bs.workload.fixtureDir(), bs.version, bs.backupProperties)
+}
+
+// getAOSTCmd returns a sql cmd that will return a system time that is greater
+// or equal than the start time of bs.incount backups in the target backup
```

Comment:
nit: s/bs.incount/the incremental backup/g

Reply:
Done.